tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/DecisionTreeTrainer.scala

```diff
@@ -229,18 +228,27 @@ class DecisionTreeTrainer(
         node.featureVectorSubset.forall(data.getNthVector(_).outcome ==
           data.getNthVector(node.featureVectorSubset.head).outcome))) {
 
-        val (featuresToExamine, featuresToIgnore) = {
+        val featuresToExamine = {
           val shuffled = Random.shuffle(node.featureSubset)
-          (shuffled.take(featuresExaminedPerNode), shuffled.drop(featuresExaminedPerNode))
+          shuffled.take(featuresExaminedPerNode)
         }
 
-        // can be expensive
-        val infoGainByFeature: Seq[(Int, Double)] =
-          computeInformationGainUnoptimized(
-            data, node.featureVectorSubset, featuresToExamine
-          ) filter {
-            case (_, gain) => gain > 0 // we get rid of features with zero information gain
-          }
+        // Computes information gain of each selected feature.
+        // Can be expensive.
+        val infoGainByFeature: Seq[(Int, Double)] = informationGainMetric match {
+          case "entropy" =>
+            computeEntropyBasedInformationGain(
+              data, node.featureVectorSubset, featuresToExamine
+            ) filter {
+              case (_, gain) => gain > 0 // we get rid of features with low information gain
+            }
+          case "multinomial" =>
+            computeMultinomialBasedInformationGain(
+              data, node.featureVectorSubset, featuresToExamine
+            ) filter {
+              case (_, gain) => gain > 1.0 // we get rid of features with low information gain
```

Comment:
Attach filter function to the gain metric? Or maybe attach threshold to it, if that makes more sense?